datalad/runner/nonasyncrunner.py

```diff
@@ -85,14 +87,26 @@ def __init__(self,
         self.return_code = None
         self.state = self.GeneratorState.process_running
         self.all_closed = False
+        self.send_lock = threading.Lock()
 
     def _check_result(self):
         self.runner._check_result()
 
-    def send(self, _):
+    def send(self, message):
+        with self.send_lock:
+            return self._locked_send(message)
+
+    def _locked_send(self, message):
+        if self.state == self.GeneratorState.initialized:
+            if message is not None:
+                raise RuntimeError(
+                    "sent non-None message to initialized generator "
```

Comment:
should it be

```suggestion
                    f"sent non-None message {message!r} to initialized generator "
```
to ease debugging etc or `message` could be too big here?